Better introduce logs and explain them (#1062)
Scalingo/documentation: _posts/platform/app/2000-01-01-logs.md

---
title: Application Logs
nav: Logs
modified_at: 2020-09-18 00:00:00
tags: app logs
index: 18
---

> Application logs can help you understand what is happening inside your application when deployed in production. Logs are especially useful for debugging problem and monitoring your application activity.

Reviewer: ```suggestion
Application logs can help you understand what is happening inside your application when deployed in production. Logs are especially useful for debugging problems and monitoring your application activity.
```

On Scalingo, everything your application writes on `stdout` and `stderr` is captured and gathered into the application logs. These logs are accessible through [Scalingo CLI]({% post_url platform/cli/2000-01-01-features %}) or through the web dashboard in the "Logs" tab:

{% assign img_url = "https://cdn.scalingo.com/documentation/screenshot_dashboard_logs.png" %}
{% include mdl_img.html %}

When opening the "Logs" tab, the last 100 lines are displayed and all new lines are automatically streaming to your browser.

If you need a bit of historical logs, you should use the Scalingo CLI with:

```shell
scalingo --app my-app logs --lines <number of log lines>
```

Replace `<number of log lines>` with the amount of lines you want to display.

> These live logs are called **hot logs**: they are the most recent logs of your application. When your application logged more than 50 MiB logs, Scalingo archive them in a log archives. We call them **cold logs**: they are not as recent as the hot logs but can be looked up for later analysis.

Reviewer: ```suggestion
These live logs are called **hot logs**: they are the most recent logs of your application. When your application logged more than 50 MiB logs, Scalingo archive them in a log archive. We call them **cold logs**: they are not as recent as the hot logs but can be looked up for later analysis.
```

> These live logs are called **hot logs**: they are the most recent logs of your application. When your application logged more than 50 MiB logs, Scalingo archive them in a log archives. We call them **cold logs**: they are not as recent as the hot logs but can be looked up for later analysis.

Reviewer 2: ```suggestion
These live logs are called **hot logs**: they are the most recent logs of your application. When your application logged more than 50 MiB logs, Scalingo archive them in a log archives. We call them **cold logs**: they are not as recent as the hot logs but can be downloaded for later analysis.
```

## Logs Archives

> All your logs are archived by Scalingo for one year. When your log file grows bigger than 50 MiB, Scalingo archive it. It is then available in the Logs Archive. Archived logs are no longer available with the `logs` command of the CLI.

Reviewer: ```suggestion
All your logs are archived by Scalingo for one year. When your log file grows bigger than 50 MiB, it is archived automatically. It is then available in the Logs Archive. Archived logs are no longer available with the `logs` command of the CLI.
```

> All your logs are archived by Scalingo for one year. When your log file grows bigger than 50 MiB, Scalingo archive it. It is then available in the Logs Archive. Archived logs are no longer available with the `logs` command of the CLI.

Reviewer 2: ```suggestion
All your logs are **archived for 1 year**. When your log file grows bigger than 50 MiB, Scalingo archive it. It is then available in the Logs Archive. Archived logs are no longer available with the `logs` command of the CLI.
```

Reviewer: It's not true for App logs, it's about router logs for these

Reviewer 2: @Soulou can you explain what is the log storage policy please?

Reviewer: Internal Connection Logs
→ Archived Internally for 1 year (for legal reason)

App Logs (which may include Connection logs if user checks the box)
→ Hot Logs:  up to 10 minutes (internally stored in Redis)
→ Cold Logs: up to 50Mib (internally stored on Disk)
→ Archives: Stored for 1 year officially (but we're not doing any garbage collection for now, no legal pressure here, it's our choice)

Reviewer 2: What are "Internal Connection Logs"? I've never seen this term

Reviewer 2: What is the policy regarding addon logs?

Reviewer: Internal Connection Logs == Router Logs

It's just that we store them separately, it doesn't depend on the fact a user has enabled Router Logs on their apps or not

Reviewer: > What is the policy regarding addon logs?

Exactly the same as App (for our log service, an addon is like an app)

Reviewer 2: Furthermore, what is the implication of activating router logs? Are the logs stored twice, once in their own storage, once mixed into the application logs?

Reviewer: > Furthermore, what is the implication of activating router logs? Are the logs stored twice, once in their own storage, once mixed into the application logs?

Yes logs are stored twice
